client/state/shortcodes/selectors.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * Returns true if currently requesting that shortcode for the specified site ID, or
+ * false otherwise.
+ *
+ * @param  {Object}  state     Global state tree
+ * @param  {Number}  siteId    Site ID
+ * @param  {String}  shortcode Shortcode
+ * @return {Boolean}           Whether that shortcode is being requested
+ */
+export const isRequestingShortcode = ( state, siteId, shortcode ) => {
+	if ( ! has( state, [ 'shortcodes', 'requesting', siteId, shortcode ] ) ) {
```

Comment:
You can be certain that at least the `state.shortcodes.requesting` part of the path will always exist, so it could save a few cycles in Lodash to avoid checking whether they're accessible.

You could also use the third argument to [`_.get`](https://lodash.com/docs/4.16.4#get) to simplify the fallback:

```js
return get( state.shortcodes.requesting, [ siteId, shortcode ], false );
```

Or:

```js
return !! get( state.shortcodes.requesting, [ siteId, shortcode ] );
```